core/src/main/java/io/fabric8/maven/core/service/openshift/ImageStreamService.java

```diff
@@ -205,23 +209,27 @@ private String findTagSha(OpenShiftClient client, String imageStreamName, String
             if (tags == null || tags.isEmpty()) {
                 continue;
             }
-            // latest tag is the first
+
+            // Iterate overall tags ang get the latest one by 'created' attribute
             TAG_EVENT_LIST:
             for (NamedTagEventList list : tags) {
                 List<TagEvent> items = list.getItems();
                 if (items == null) {
                     continue TAG_EVENT_LIST;
                 }
 
-                // latest item is the first
                 for (TagEvent item : items) {
```

Comment:
Have you considered simplifying the iterations and filtering using java8 stream?